apps/graphql/src/apps/itinerary/Itinerary.js

```diff
@@ -15,16 +15,17 @@ type Passengers = {|
   +infants?: number,
 |};
 
-type SearchLocation = {|
-  +ids: string[],
+export type SearchLocation = {|
+  +ids?: string[],
+  +slugs?: string[],
 |};
 
 export type ItinerariesOneWaySearchParameters = {|
   +order?: Order,
   +sort?: Sort,
   +itinerary: {|
-    +origin: SearchLocation,
-    +destination?: SearchLocation,
+    origin: SearchLocation,
+    destination?: SearchLocation,
```

Comment:
Why is this not read-only?

Reply:
Because we are reassigning them to parse the IDs on `fetchItineraries` 

```
params.itinerary.origin = slugsNeedParsing(origin)
        ? await parseSlugsToIds(origin)
        : unmaskReplaceIds(origin);
```
If I put it into different variables and then compose them to pass them to `parseParameters` get a lot of weird Flow errors...